packages/blockchain-link/src/workers/ripple/index.ts

```diff
@@ -190,30 +218,27 @@ const getTransaction = async ({ connect, payload }: Request<MessageTypes.GetTran
 };
 
 const pushTransaction = async ({ connect, payload }: Request<MessageTypes.PushTransaction>) => {
-    const api = await connect();
+    const client = await connect();
     // tx_blob hex must be in upper case
-    const info = await api.submit(payload.toUpperCase());
+    const info = await client.submit(payload.toUpperCase());
 
-    if (info.resultCode === 'tesSUCCESS') {
+    if (info.result.engine_result === 'tesSUCCESS' && info.result.tx_json.hash) {
         return {
             type: RESPONSES.PUSH_TRANSACTION,
-            // payload: info.resultMessage,
-            // @ts-expect-error this param is not typed in RippleApi
-            payload: info.tx_json.hash,
+            payload: info.result.tx_json.hash,
         } as const;
     }
-    throw new Error(info.resultMessage);
+    throw new Error(info.result.engine_result_message);
 };
 
 const estimateFee = async (request: Request<MessageTypes.EstimateFee>) => {
-    const api = await request.connect();
-    const fee = await api.getFee();
-    // TODO: sometimes rippled returns very high values in "server_info.load_factor" and calculated fee jumps from basic 12 drops to 6000+ drops for a moment
-    // investigate more...
```

Comment:
Might be worth keeping this just in case

Reply:
We have also condition in `MiscFees.ts` so I am not sure if we still need.

Reply:
ok we need it
```
const RIPPLE_FEE_INFO: FeeInfoWithLevels = {
    blockTime: -1, // unknown
    defaultFees: [{ label: 'normal', feePerUnit: '12', blocks: -1 }],
    minFee: 10,
    maxFee: 10000,
    dustLimit: -1, // unknown/unused
};

```